apps/src/netsim/NetSimHeartbeat.js

```diff
@@ -133,12 +150,34 @@ NetSimHeartbeat.prototype.setBeatInterval = function (intervalMs) {
 };
 
 /**
+ * Set a handler to call if this heartbeat is unable to update its remote
+ * storage representation.  Can be used to go into a recovery mode,
+ * acknowledge disconnect, and/or attempt an auto-reconnect.
+ * @param {function} onFailedHeartbeat
+ * @throws if set would clobber a previously-set callback
+ */
+NetSimHeartbeat.prototype.setFailureCallback = function (onFailedHeartbeat) {
+  if (this.onFailedHeartbeat_ !== undefined && onFailedHeartbeat !== undefined) {
+    throw new Error("Heartbeat already has a failure callback.");
+  }
+  this.onFailedHeartbeat_ = onFailedHeartbeat;
+};
+
+/**
  * Updates own row on regular interval, as long as something's making
  * it tick.
  */
 NetSimHeartbeat.prototype.tick = function () {
   if (Date.now() - this.time_ > this.intervalMs_) {
     this.time_ = Date.now();
-    this.update();
+    this.update(function (err/*, result*/) {
```

Comment:
Not sure if there's any value in this commented out result. I'd either just leave it in there uncommented (and let minifier remove it) or just remove it entirely.